src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -259,7 +259,8 @@ public async Task<ServerNode> GetCurrentSessionNode()
                 extensions:
                 [
                     JavascriptConversionExtensions.LinqMethodsSupport.Instance,
-                    JavascriptConversionExtensions.NullableSupport.Instance
+                    JavascriptConversionExtensions.NullableSupport.Instance,
+                    JavascriptConversionExtensions.PatchDictionaryEnumSupport.Instance,
```

Comment:
Why do we _always_ (regardless `SaveEnumsAsIntegersForPatching` value) add the enum extension using `EnumOptions.UseStrings`? (I don't entirely follow your previous answer - https://github.com/ravendb/ravendb/pull/18530#discussion_r1613574754)

I thought that what we'll need here will be more like:

```
var extensions = new List { 
                     JavascriptConversionExtensions.LinqMethodsSupport.Instance,
                    JavascriptConversionExtensions.NullableSupport.Instance
}

if (_convensions.SaveEnumsAsIntegersForPatching)
    extensions.Add(EnumConversionSupport.EnumsAsIntegersInstance);
```

Reply:
I also thought that at first, but there is another bug/feature at play here.

`SaveEnumsAsIntegers` does not, and never have, changed dictionary enum keys to integers when **storing**. So the easy way out is to modify patching to also behave like that when genereting the js/jsonpath for patches.

So if we change to use integers for dictionary enum keys here we are again breaking an existing behaviour *or* creating other inconsitensies between store and patch.

The correct fix would be to change `SaveEnumsAsIntegers` to actually use integers for storing enum keys but that would be an even bigger breaking change which cannot be fixed by `SaveEnumsAsIntegersForPatching` because it's about storing and not patching.